### PR TITLE
Manage a list field by allowing the <$select> widget to select multiple items.

### PR DESCRIPTION
**Pull request by mklauber**
As a means to manage a list field, allow a <$select> widget to display and select multiple items.  For each item in the list field, select the item in the listbox.  Each time the selected items in the list box change, update the items in the list field.

### CODE REVIEW

File: core/modules/widgets/select.js [hunk 1]
```diff
@@ -51,7 +51,12 @@ SelectWidget.prototype.render = function(parent,nextSibling) {
 Handle a change event
 */
 SelectWidget.prototype.handleChangeEvent = function(event) {
-	var value = this.getSelectDomNode().value;
+	if(this.selectMultiple == false) {
+		var value = this.getSelectDomNode().value;
+	} else {
+		var value = this.getSelectValues()
```
Comment (pmario) [inline core/modules/widgets/select.js]: hmmm, ... I know, we have a very relaxed variable declaration handling in TW, but imo this one hurts. see line 55 and 57. @Jermolene ??

File: licenses/cla-individual.md [hunk 2]
```diff
@@ -256,3 +256,5 @@ Tony Grosinger @tgrosinger 2015/10/03
 Antaeus Feldspar @afeldspar 2015/10/20
 
 Soeren Enevoldsen, @senevoldsen90, 2015/10/09
+
+Matthew LAuber, @mklauber, 2015/11/13
```
Comment (pmario) [inline licenses/cla-individual.md]: typo?

**Comment by pmario** [comment]
What's about the order of the values. ... Do they correlate to the order in the select dropdown?

File: core/modules/widgets/select.js (hunk 1 above)
Comment (mklauber) [inline core/modules/widgets/select.js]: Would you prefer this? 
```javascript
SelectWidget.prototype.handleChangeEvent = function(event) {
	var value = this.selectMultiple === false ? this.getSelectDomNode().value :
												$tw.utils.stringifyList(this.getSelectValues());
	this.wiki.setText(this.selectTitle,this.selectField,this.selectIndex,value);
};
```

**Comment by mklauber** [comment]
Yes.  It will iterate in order over the list, and add the text to the list field in that order.  If you want to change the list field order, you need to change the select widget order.

**Comment by mklauber** [comment]
And if you edit the list field manually, this will overwrite the changes.

**Comment by mklauber** [comment]
Also, One thing I just thought of.  If you allow duplicate values in the select widget, you could get duplicate values in the list field.  Not sure what the correct behavior would be there.

**Comment by tobibeer** [comment]
Hi @mklauber,

Please take out the CLA and make a signed PR for it separately.
Any demo somewhere?

File: core/modules/widgets/select.js (hunk 1 above)
Comment (tobibeer) [inline core/modules/widgets/select.js]: I imagine the convention for the tw repo to be:

```
var value;
if(this.selectMultiple == false) {
	value = this.getSelectDomNode().value;
} else {
	value = this.getSelectValues()
}
```

Personally, I'd prefer:

```
var value = this.selectMultiple == false ?
	this.getSelectDomNode().value :
	this.getSelectValues();
```

Comment (pmario) [inline core/modules/widgets/select.js]: noooo. way to complicated and error prone to read and maintain!

I'd prefer something like this:

```
SelectWidget.prototype.handleChangeEvent = function(event) {
    var value;
    if () {
        value = ...
    }
```

File: core/modules/widgets/select.js [hunk 3]
```diff
@@ -94,6 +111,22 @@ SelectWidget.prototype.getSelectDomNode = function() {
...
 };
 
+// Return an array of the selected opion values
+// select is an HTML select element
+SelectWidget.prototype.getSelectValues = function() {
+	var select = this.getSelectDomNode()
+	var result = [];
+	var options = select && select.options;
+	var opt;
+	for (var i=0, iLen=options.length; i<iLen; i++) {
```
Comment (tobibeer) [inline core/modules/widgets/select.js]: I think `iLen` needs to be declared too? Perhaps just use length.

**Comment by tobibeer** [comment]
> Also, One thing I just thought of. If you allow duplicate values in the select widget, you could get duplicate values in the list field. Not sure what the correct behavior would be there.

Then that would be "works" as designed and write duplicates.

**Comment by mklauber** [comment]
> Any demo somewhere? :-)

http://www.mklauber.com/listselect.html

File: core/modules/widgets/select.js (hunk 1 above)
Comment (mklauber) [inline core/modules/widgets/select.js]: I'll update this however we'd like, as soon as there's a consensus.  @Jermolene, oh BDFL, will you bring clarity to our muddled musings? :-P

Comment (mklauber) [inline core/modules/widgets/select.js]: @Jermolene ping?

**Comment by tobibeer** [comment]
:+1:  ...and , did I meantion? Good stuff. ;-)

**Comment by Jermolene** [comment]
Thanks @mklauber, and apologies for the delayed merge.

**Pull request merged**